src/Accounts/Authentication.ResourceManager/ProtectedFileProvider.cs

```diff
@@ -87,18 +92,17 @@ public StreamWriter CreateWriter()
 
         protected virtual void InitializeStream()
         {
-            lock (_initializationLock)
+            _initializationLock.WaitOne();
+            if (_stream == null)
             {
-                if (_stream == null)
+                Stream stream;
+                if (!TryGetStreamLock(AcquireLock, FilePath, out stream))
                 {
-                    Stream stream;
-                    if (!TryGetStreamLock(AcquireLock, FilePath, out stream))
-                    {
-                        throw new UnauthorizedAccessException(string.Format(Resources.FileLockFailure, FilePath));
-                    }
-
-                    _stream = stream;
+                    _initializationLock.ReleaseMutex();
+                    throw new UnauthorizedAccessException(string.Format(Resources.FileLockFailure, FilePath));
                 }
+                _initializationLock.ReleaseMutex();
```

Comment:
how about put `_initializationLock.ReleaseMutex()` in finally block?

Reply:
Makes sense. Done